src/tracer.rs

```diff
@@ -101,9 +98,36 @@ impl ObjectSubclass for TracingTracerPriv {
         }
     }
 }
+pub(crate) trait TracingTracerImpl: TracerImpl {}
+
+unsafe impl<T: TracingTracerImpl> IsSubclassable<T> for TracingTracer {
+    fn class_init(class: &mut glib::Class<Self>) {
+        Self::parent_class_init::<T>(class);
+    }
+}
 
 impl ObjectImpl for TracingTracerPriv {
     fn constructed(&self) {
+        if let Some(params) = self.obj().property::<Option<String>>("params") {
+            let tmp = format!("params,{}", params);
+            info!("{:?} params: {:?}", self.obj(), tmp);
+            let structure = gstreamer::Structure::from_str(&tmp).unwrap_or_else(|e| {
+                error!("Invalid params string: {:?}: {e:?}", tmp);
+                gstreamer::Structure::new_empty("params")
+            });
+
+            if let Ok(gst_logs_level) = structure
+                .get::<String>("log-level")
+                .or_else(|_| structure.get::<i32>("log-level").map(|l| l.to_string()))
+            {
+                info!("Integrating `{gst_logs_level}` GStreamer logs as part of our tracing");
+
+                crate::integrate_events();
```

Comment:
Hm, won’t this result in infinite recursion, given that `integrate_spans` creates a new instance of `TracingTracer`?